src/ILLink.RoslynAnalyzer/RequiresUnreferencedCodeAnalyzer.cs

```diff
@@ -121,5 +117,21 @@ internal static bool IsNamedType (INamedTypeSymbol type, string typeName)
 
 			return true;
 		}
+
+		/// <summary>
+		/// Returns a RequiresUnreferencedCodeAttribute if found
+		/// </summary>
+		static AttributeData? GetRequiresUnreferencedCodeAttribute (ImmutableArray<AttributeData> attributes)
```

Comment:
nit: `RequiresUnreferencedCode` can only be placed in methods, so why not limit this to `IMethodSymbol`? We are already checking this only for methods since we only call this from within `CheckMethodOrCtorCall`. We could also change this to a `TryGet...` and return the attribute in an `out` and then use this above as `if (TryGet...(method, out requiresUnreferncedCode) { ... }`.